### DISCUSSION TITLE
Thread pool max concurrency option

### DISCUSSION THREAD

**Discussion by valsov** [elsa-workflows/elsa-core#1308] [Q&A] [answered]
Hi !

My use case of Elsa will often cause about 200 workflow instances to execute at the same time. This may be a problem in case of memore/cpu intensive activities.
I was previously using Quartz.Net to execute those jobs (now converted to workflows) and was able to configure the ThreadPool with a MaxConcurrency.
```cs
services.AddQuartz(q =>
{
    [...]
    q.UseDefaultThreadPool(x => x.MaxConcurrency = configuration.MaxThreads);
    [...]
});
```
Is there a similar configuration built into Elsa ? I looked around and found no clue about this.

Thank you for your time

**Comment by sfmskywalker** [comment, marked as answer]
Hi @valsov,

Elsa itself doesn't have any workers to configure, because these are delegated to its various providers. For example, timer-based events might be handled by either Quartz.NET or Hangfire, depending on which one you configure. Consequently, you can configure these providers yourself with things like `MaxConcurrency` in the case of Quartz.NET and worker count for Hangfire.

Example when using Elsa Temporal activities using Quartz:

```csharp
services.AddElsa(elsa => elsa.AddQuartzTemporalActivities(configureQuartz: quartz => quartz.UseDefaultThreadPool(x => x.MaxConcurrency = 4));
```

Another example is the default queue worker for processing workflow execution jobs happening in the background. Elsa uses Rebus to send and process messages, so you can configure Rebus with Rebus-specific options. Example:

```csharp
services.AddElsa(elsa => 
{
   elsa.ElsaOptions.ServiceBusOptions.NumberOfWorkers = 10;
   elsa.ElsaOptions.ServiceBusOptions.MaxParallelism = 10;
}));
```

**Reply by valsov** [reply]
Thank you, that's exactly what I was looking for.
Have a nice day

**Reply by kelvenray** [reply]
Thanks.That's what I want exactly as well.
